helix-core/src/test/java/org/apache/helix/controller/stages/TestRecoveryLoadBalance.java

```diff
@@ -57,7 +57,10 @@
   private final String STATE_MODEL = "statemodel";
   private ClusterConfig _clusterConfig;
 
-  @Test(dataProvider = "recoveryLoadBalanceInput")
+  // Disable the test since recovery rebalance is not able to block load rebalance.
+  // TODO: if we decide to support that by replica level, we will have a different config for that and this test can be
```

Comment:
This is the pain point that we want to address with this change. I suggest change the test to validate the reverse way. So you don't have to disable the test, and the new change will be protected by more coverage.

Reply:
I will have a new test class already covered most of cases.

Disable the test is keep the possibility of supporting the feature again. As I said, this feature is not determined to implement or not.

Reply:
I would suggest removing this test given we already have test coverage. First, there is no plan or discussion about re-enabling it. Secondly, I think simply block loadbalance because of recovery without considering throttling is not correct behavior. If we are going to do a similar restriction, we would do it through throttling anyway.